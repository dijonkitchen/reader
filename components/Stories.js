import React from 'react';
import {
  StyleSheet,
  ListView,
} from 'react-native';

import ShortStory from './ShortStory';

export default class Stories extends React.Component {
  constructor() {
    super();
    const ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1 !== r2,
    });
    this.state = {
      stories: ds.cloneWithRows([]),
    };
  }

  componentDidMount() {
    this.queryFeed();
  }

  queryFeed() {
    fetch("https://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20rss%20where%20url%3D'http%3A%2F%2Ffeeds.reuters.com%2Freuters%2FMostRead'&format=json&env=store%3A%2F%2Fdatatables.org%2Falltableswithkeys")
      .then(response => response.json())
      .then((responseJson) => {
        const mostRead = responseJson.query.results.item;
        // console.log(mostRead[0])
        this.setState({
          stories: this.state.stories.cloneWithRows(mostRead),
        });
      })
      .catch((error) => {
        console.error(error)
      });
  }

  render() {
    return (
      <ListView
        style={styles.container}
        dataSource={this.state.stories}
        renderRow={(rowData) => {
          return (
            <ShortStory
              data={rowData}
              navigation={this.props.navigation}
            />
          );
        }}
      />
    );
  }
}

Stories.navigationOptions = {
  title: 'Stories',
};

const styles = StyleSheet.create({
  container: {
    margin: 15,
  },
});
