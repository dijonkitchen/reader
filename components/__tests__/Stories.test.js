import React from 'react';
import { ListView, View } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

import Stories from '../Stories';

describe('Stories component', () => {
  beforeEach(() => {
    wrapper = shallow(<Stories />);
  });

  test('render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should be a ListView component', () => {
    expect(wrapper.type()).toEqual(ListView);
  });

  test('renders correctly', () => {
    const tree = renderer.create(
      <Stories />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
