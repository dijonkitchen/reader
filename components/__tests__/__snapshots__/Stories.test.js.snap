// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories component renders correctly 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 0,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={
    Object {
      "margin": 15,
    }
  }
>
  <View />
</RCTScrollView>
`;
